crates/generate-raw/src/lib.rs

```diff
@@ -223,6 +241,14 @@ impl Render for BuiltinType {
             BuiltinType::S64 => src.push_str("i64"),
             BuiltinType::F32 => src.push_str("f32"),
             BuiltinType::F64 => src.push_str("f64"),
+            BuiltinType::USize => {
+                // TODO verify handling of USize
+                src.push_str("usize")
+            }
+            BuiltinType::Char8 => {
+                // TODO verify handling of Char8
+                src.push_str("char")
```

Comment:
I don't believe this is correct - std::mem::size_of::<char>() is 4, but according to witx's layout implementation, Char8's size is 1.

Reply:
Oh bummer, you’re absolutely right! I’ll submit a fix then. Would you agree that changing it to `u8` would work?

Reply:
But then, since we operate on UTF8 in WASI and `Char8` is one byte, does that mean we’ll potentially need a couple to represent a valid UTF8 codepoint? Or am I missing something here?

Reply:
So, I am not confident in my string encoding expertise, but in wasi ephemeral (the only place Char8 is used so far) it is only used as the target of a pointer `(@witx pointer char8)` for the callee to provide a buffer for the caller to write a string to. The string encoding is UTF-8, but since a codepoint can span multiple bytes, we can make no guarantees about an individual Char8.

So, I think we should treat it as a rust `u8` in this setting, not a `char`. The user of these bindings can turn their `&str` into `&[u8]` with `.as_bytes()`, and then split that slice into a pointer and a length with `.as_ptr()` and `.len()`.

Reply:
Yes, treating it as `u8` is correct. `char8` is meant to correspond to C++20's `char8_t`. It's a UTF-8 code *unit* (`u8` in Rust), rather than a code *point* (`char` in Rust).

In isolation, a `char8` acts just like a `u8`. The main reason for having `char8` as a separate type is that it documents which arrays we expect to be valid UTF-8 instead of just arbitrary bytes. However, we still enforce UTF-8 dynamically, at each call, rather than through static type checking.

Reply:
Right, `char8` being a code *unit* rather than a code *point* makes sense. I'll provide a fix ASAP then. Thanks!